openmc/mgxs/groups.py

```diff
@@ -77,6 +77,76 @@ def group_edges(self, edges):
         cv.check_greater_than('number of group edges', len(edges), 1)
         self._group_edges = np.array(edges)
 
+    @staticmethod
+    def get_common_energy_groups(name):
+        """Returns one of many commonly used energy group structures.
+
+        Pre-defined structures include "CASMO-X" (where X is 2, 4, 8, 16, 25,
+        40 or 70) from the [CASMO]_ lattice physics code.
+
+        References
+        ----------
+        .. [CASMO] J. Rhodes, K. Smith, and D. Lee, "CASMO-5 Development and
+           Applications", *Proc. ANS Topical Meeting on Reactor Physics* (2006).
+
+        Parameters
+        ----------
+        name : str
+            The name of the energy group structure.
+
+        Returns
+        -------
+        EnergyGroups
+            The energy group structure 
+
+        """
+
+        if name == "CASMO-2":
+            group_edges = np.array([
+                0., 6.25e-1, 2.e7])
+        elif name == "CASMO-4":
+            group_edges = np.array([
+                0., 6.25e-1, 5.53e3, 8.21e5, 2.e7])
+        elif name == "CASMO-8":
+            group_edges = np.array([
+                0., 5.8e-2, 1.4e-1, 2.8e-1, 6.25e-1, 4., 5.53e3, 8.21e5, 2.e7])
+        elif name == "CASMO-16":
+            group_edges = np.array([
+                0., 3.e-2, 5.8e-2, 1.4e-1, 2.8e-1, 3.5e-1, 6.25e-1, 8.5e-1,
+                9.72e-1, 1.02, 1.097, 1.15, 1.3,  4., 5.53e3, 8.21e5, 2.e7])
+        elif name == "CASMO-25":
+            group_edges = np.array([
+                0., 3.e-2, 5.8e-2, 1.4e-1, 2.8e-1, 3.5e-1, 6.25e-1, 9.72e-1,
+                1.02, 1.097, 1.15, 1.855, 4., 9.877, 1.5968e1, 1.4873e2,
+                5.53e3, 9.118e3, 1.11e5, 5.e5, 8.21e5, 1.353e6, 2.231e6,
+                3.679e6, 6.0655e6, 2.e7])
+        elif name == "CASMO-40":
+            group_edges = np.array([
+                0., 1.5e-2, 3.e-2, 4.2e-2, 5.8e-2, 8.e-2, 1.e-1, 1.4e-1,
+                1.8e-1, 2.2e-1, 2.8e-1, 3.5e-1, 6.25e-1, 8.5e-1, 9.5e-1,
+                9.72e-1, 1.02, 1.097, 1.15, 1.3, 1.5, 1.855, 2.1, 2.6, 3.3, 4.,
+                9.877, 1.5968e1, 2.77e1, 4.8052e1, 1.4873e2, 5.53e3, 9.118e3,
+                1.11e5, 5.e5, 8.21e5, 1.353e6, 2.231e6, 3.679e6, 6.0655e6,
+                2.e7])
+        elif name == "CASMO-70":
+            group_edges = np.array([
```

Comment:
Just found three values may be typos when comparing the 70-group structure with the one I often use. I think the correct values should be:
```
group_edges_70[30] = 9.96e-1
group_edges_70[51] = 9.069e2
group_edges_70[56] = 9.118e3
```
All the other values are the same as I checked.
Again, welcome back! @wbinventor

Reply:
Thanks for double checking the values @johnnyliu27! I'll update the values this evening. Oddly enough, I do think that you and I are either using different structures, or the typo originated in the structure I used for my thesis. Which version of CASMO did your structures come from? I know that the structures changed slightly between v4 and v5; mine (supposedly) came from v5.